src/librustc/dep_graph/dep_tracking_map.rs

```diff
@@ -134,7 +175,7 @@ impl<M: DepTrackingMapConfig> MemoizationMap for RefCell<DepTrackingMap<M>> {
     /// accesses the body of the item `item`, so we register a read
     /// from `Hir(item_def_id)`.
     fn memoize<OP>(&self, key: M::Key, op: OP) -> M::Value
-        where OP: FnOnce() -> M::Value
+        where OP: FnOnce() -> Option<M::Value>
```

Comment:
it looks to me like there is no longer a need for this part of the diff -- that is, we can make `memoize()` take a closure that returns just `M::Value` again